### PR TITLE
[teensy_loader_cli] Add teensy_loader_cli

### PR DESCRIPTION
**Pull request by Seelengrab**
This depends on https://github.com/JuliaPackaging/Yggdrasil/pull/7320.

To any future people that want to bring this to non-linux systems - [read the Makefile](https://github.com/PaulStoffregen/teensy_loader_cli/blob/master/Makefile), and godspeed.

### CODE REVIEW

**Comment by Seelengrab** [comment]
Hm. I had hoped reopening would restart CI, since the `libusb_compat_jll` is available now :thinking:

**Comment by giordano** [comment]
Buildkite isn't triggered by changes on the merge commit, only of the tip of the branch, so closing/reopening the PR doesn't have any effect (sadly).  They seem to consider this a feature :shrug:

**Comment by Seelengrab** [comment]
Yeah, I can understand why - prevents misuse be constantly closing & reopening a PR. But then again, people could just automate creating PRs.. ah well :shrug:

File: T/teensy_loader_cli/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/PaulStoffregen/teensy_loader_cli", "99082869db86f1f5ff7eef0d45262bc7e674f890")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/teensy_loader_cli
+make -j${nprocs}
+mv teensy_loader_cli $WORKSPACE/destdir/bin/teensy_loader_cli
```
Comment (giordano) [inline T/teensy_loader_cli/build_tarballs.jl:18]:
```suggestion
install -Dvm 755 "teensy_loader_cli${exeext}" "${bindir}/teensy_loader_cli${exeext}"
```

**Pull request merged**